### PR TITLE
Add/edit posts tests

### PR DESCRIPTION
**Pull request by tellthemachines**
Sequel to #200, fixes element targeting issues in some of the tests.

Trac ticket: https://core.trac.wordpress.org/ticket/49507

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/e2e/specs/edit-posts.test.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	pressKeyTimes,
+	switchUserToAdmin,
+	switchUserToTest,
+	visitAdminPage,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Trash any posts on the edit posts screen.
+ */
+async function trashExistingPosts() {
```
Comment (talldan) [inline tests/e2e/specs/edit-posts.test.js:12]: It wasn't there when the code was originally written, but this should be able to use `trashAllPosts` from test utils now:
https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-test-utils/src/posts.js#L21

Comment (tellthemachines) [reply tests/e2e/specs/edit-posts.test.js:12]: Hmm, I'm trying to use it but it errors with `invalid post type`. The default is `post` so I'm not sure what's happening 😕

Comment (tellthemachines) [reply tests/e2e/specs/edit-posts.test.js:12]: Update: in order to get `trashAllPosts` to work I have to call it within a callback passed to `beforeEach`, but when I try that the admin and password strings get bizarrely mangled:

Screen Shot 2021-03-22 at 1 23 54 pm

Comment (talldan) [reply tests/e2e/specs/edit-posts.test.js:12]: Interesting, that sounds like what was happening when I tested the visual regression tests:
https://github.com/WordPress/wordpress-develop/pull/209#pullrequestreview-573957665

That gif doesn't show it, but I did see the tests unable to type properly in those input fields.

Comment (tellthemachines) [reply tests/e2e/specs/edit-posts.test.js:12]: Found the issue: I forgot to `await` `trashAllPosts` 🤦

**Review by youknowriad — COMMENTED** [review]
Great work here. Love seeing e2e test added to Core. It's important though to not add Gutenberg e2e tests to Core because we already have a way to run Gutenberg tests in Core.

I think this one is ready for commit

**Comment by JustinyAhin** [comment]
I've run the tests, and they work as expected.

@tellthemachines I was wondering, does it make sense to add a README to the e2e folder with information about how to run the test, how to enable headless mode for the tests for instance?

Also, what does this needs to get merged?

**Comment by talldan** [comment]
There's a README over here @JustinyAhin - https://github.com/WordPress/wordpress-develop/pull/1451.

**Comment by hellofromtonya** [comment]
@tellthemachines @talldan Is this PR ready for `commit`? Does it overlap any of the work that @JustinyAhin is doing?

**Comment by tellthemachines** [comment]
@hellofromtonya yup, this one's ready to go. It doesn't overlap with Justin's PRs as this is specific to the posts screen and Justin has been focusing on other screens so far.

**Comment by hellofromtonya** [comment]
When running this test locally (via `npm run test:e2e`), I get the following error:

```
 FAIL  tests/e2e/specs/edit-posts.test.js (76.359 s)
  ● Edit Posts › allows an existing post to be edited using the Edit button

    TimeoutError: waiting for selector `.components-snackbar` failed: timeout 30000ms exceeded

      44 | 		const title = 'Test Title';
      45 | 		await createNewPost( { title } );
    > 46 | 		await publishPost();
         | 		^
      47 | 		await visitAdminPage( '/edit.php' );
      48 |
      49 | 		await page.waitForSelector( '#the-list .type-post' );

      at new WaitTask (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:780:28)
      at DOMWorld.waitForSelectorInPage (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:632:22)
      at Object.internalHandler.waitFor (../../node_modules/puppeteer-core/src/common/QueryHandler.ts:78:19)
      at DOMWorld.waitForSelector (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:490:25)
      at Frame.waitForSelector (../../node_modules/puppeteer-core/src/common/FrameManager.ts:1163:47)
      at Page.waitForSelector (../../node_modules/puppeteer-core/src/common/Page.ts:2100:29)
      at publishPost (../../node_modules/@wordpress/e2e-test-utils/build/@wordpress/e2e-test-utils/src/publish-post.js:19:14)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (specs/edit-posts.test.js:46:3)

  ● Edit Posts › allows an existing post to be deleted using the Trash button

    TimeoutError: waiting for selector `.components-snackbar` failed: timeout 30000ms exceeded

      110 | 		const title = 'Test Title';
      111 | 		await createNewPost( { title } );
    > 112 | 		await publishPost();
          | 		^
      113 | 		await visitAdminPage( '/edit.php' );
      114 |
      115 | 		await page.waitForSelector( '#the-list .type-post' );

      at new WaitTask (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:780:28)
      at DOMWorld.waitForSelectorInPage (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:632:22)
      at Object.internalHandler.waitFor (../../node_modules/puppeteer-core/src/common/QueryHandler.ts:78:19)
      at DOMWorld.waitForSelector (../../node_modules/puppeteer-core/src/common/DOMWorld.ts:490:25)
      at Frame.waitForSelector (../../node_modules/puppeteer-core/src/common/FrameManager.ts:1163:47)
      at Page.waitForSelector (../../node_modules/puppeteer-core/src/common/Page.ts:2100:29)
      at publishPost (../../node_modules/@wordpress/e2e-test-utils/build/@wordpress/e2e-test-utils/src/publish-post.js:19:14)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (specs/edit-posts.test.js:112:3)

 PASS  tests/e2e/specs/dashboard.test.js
 PASS  tests/e2e/specs/hello.test.js

Test Suites: 1 failed, 2 passed, 3 total
Tests:       2 failed, 6 passed, 8 total
Snapshots:   0 total
Time:        79.215 s
```

@tellthemachines Does it pass locally for you?

**Comment by tellthemachines** [comment]
@hellofromtonya it passes for me! I rebased the branch in case it helps at all.

**Comment by hellofromtonya** [comment]
Committed via changeset https://core.trac.wordpress.org/changeset/51990.

**Pull request closed without merging**